### DISCUSSION TITLE
How to properly write a table name migration

### DISCUSSION THREAD

**Discussion by lpotthast** [SeaQL/sea-orm#802] [Q&A] [answered]
Lets say I defined an entity like this:

    #[sea_orm(table_name = "Courses")]
    pub struct Model {
        ...
    }

And I created a migration to create a table for these entities with
```
manager
            .create_table(
                Table::create()
                    .table(course::Entity)
                    ...
```
That table will have the name `Courses`.

Lets say I want to rename this table to its singular name `Course`. This is how my migration initially looked like
```
manager
            .rename_table(
                Table::rename()
                    .table(course::Entity, Alias::new("Course"))
                    .to_owned(),
            )
            .await
```

Executing this works, but every access to the entity than immediately complaints that the table `Courses`, which I still have in the Model annotation (`#[sea_orm(table_name = "Courses")]`), no longer exists..

Changing the annotation to ` #[sea_orm(table_name = "Course")]` *before* running the above rename migration leads to the migration failing, as it cannot find a table named `Course` at this point...

So the only way to make that rename operation work is by simply using
```
 Table::rename()
                    .table(Alias::new("Courses"), Alias::new("Course"))
```
and not relying on the type definition at all.
Am I right with this assumption? Might be something for the documentation...

**Comment by tyt2y3** [comment, marked as answer]
Sorry for the confusion, we now updated the examples.
https://github.com/SeaQL/sea-orm/pull/785

Basically you can do something like

```rust
#[derive(Iden)]
enum Post {
    Table,
    Id,
    Title,
    Text,
}
```

**Comment by gplibs** [comment]
``` rust
struct TableName(String);
impl sea_orm::Iden for TableName {
    fn unquoted(&self, s: &mut dyn fmt::Write) {
        write!(s, "{}", self.0).unwrap();
    }
}


let mut stmt = schema.create_table_from_entity(course::Entity);
// Change the table name here
stmt.table(TableName("new_table_name".to_owned()));
```
